common/HeartbeatMap: return earlier when checking for healthy

wonzhq opened the pull request:
Can return earlier if 'healthy' is false.

Signed-off-by: Zhiqiang Wang <zhiqiang.wang@intel.com>

dachary commented:
@wonzhq could you explain in the commit message what is the benefit of this change ? I assume it's aesthetic but maybe not ?

dachary commented on src/common/HeartbeatMap.cc:
```diff
@@ -121,6 +121,9 @@ bool HeartbeatMap::is_healthy()
     ldout(m_cct, 0) << "is_healthy = false, injected failure for next " << (m_inject_unhealthy_until - now) << " seconds" << dendl;
     healthy = false;
+    m_rwlock.put_read();
+    ldout(m_cct, 20) << "is_healthy = " << (healthy ? "healthy" : "NOT HEALTHY") << dendl;
```
healthy is always false here

wonzhq commented:
@dachary the benefit of this change is that if we find an un-healthy 'worker', there is no need to check the remaining 'workers'. I've updated the commit message, also updated with your comment. Thank you for taking a look!

dachary commented on src/common/HeartbeatMap.cc:
```diff
@@ -121,6 +121,9 @@ bool HeartbeatMap::is_healthy()
+    m_rwlock.put_read();
+    ldout(m_cct, 20) << "is_healthy = NOT HEALTHY" << dendl;
+    return healthy;
   }
```
what about just adding a **} else {** here so that the loop is run only if **now >= m_inject_unhealthy_until** ? It's less change and does the same. Maybe easier to read as well ?

wonzhq replied:
I'm ok to make the change if you'd like to.

liewegas commented:
This avoid some extra work, but I think ti's more important to see the messages in the logs about *all* threads that are stalled.  We don't really care about saving cycles in that case...
